File: Intermediate/class.py
class vehicle():
    def __init__(self, make,model):
        self.make=make
        self.model=model
    def start(self):
        return f"{self.model} is from {self.make}"
class car(vehicle):
    def __init__(self, make, model,price):
        super().__init__(make, model) #to inherit from the parent
        self.price=price
    def drive(self):
        return f"{self.make} {self.model} {self.price}"
c=car("TOYOTA","COROLLA",250000)
print(c.price)
print(c.model)
print(c.make)

    

File: Intermediate/class.js
//inheritance of the class

class Vehicle
{
    constructor(make, model)
    {
        this.make = make;
        this.model = model;
    }

    start()
    {
        return `${this.model} is a car from ${this.make}`;
    }
}

class Car extends Vehicle //inheritance via extends
{
    constructor(make, model, price)
    {
        super(make, model);   // ✅ call parent constructor
        this.price = price;
    }

    drive()
    {
        return `${this.make}: this is the inheritance example`;
    }
}

let mycar = new Car("Toyota", "Corolla", 10000000);

console.log(mycar);
console.log(mycar.start());
console.log(mycar.drive());



